python/pyspark/sql/connect/dataframe.py

```diff
@@ -503,6 +503,63 @@ def _show_string(
         assert pdf is not None
         return pdf["show_string"][0]
 
+    def withColumns(self, colsMap: Dict[str, Expression]) -> "DataFrame":
```

Comment:
shall we also add the `withColumns` overload that takes 2 lists?

Reply:
There is no such API in PySpark so I think probably no?